<?xml version="1.0" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from /home/oliver/git/3d-soft-trunk/urdf/2segment.urdf.xacro | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<!-- This file has been generated automatically from SoftTrunkModel::generateRobotURDF(), do not edit by hand -->
<robot name="2segment" xmlns:xacro="http://www.ros.org/wiki/xacro">
	<material name="green">
		<color rgba="0.2 1 0.2 0.8"/>
	</material>
	<material name="white">
		<color rgba="0.5 0.5 0.5 1"/>
	</material>
	<link name="base_link">
		<!--since the RBDL needs a inertial info for each link, just assign a really small value to links where we don't have to consider inertia.-->
		<inertial>
			<mass value="0.00001"/>
			<inertia ixx="0.00001" ixy="0.0000" ixz="0.0000" iyy="0.00001" iyz="0.0000" izz="0.00001"/>
		</inertial>
	</link>
	<joint name="base_link_softTrunk_base_joint" type="fixed">
		<parent link="base_link"/>
		<child link="softTrunk_base"/>
		<origin rpy="0 3.1415926499999998 0"/>
	</joint>
	<link name="softTrunk_base">
		<!--since the RBDL needs a inertial info for each link, just assign a really small value to links where we don't have to consider inertia.-->
		<inertial>
			<mass value="0.00001"/>
			<inertia ixx="0.00001" ixy="0.0000" ixz="0.0000" iyy="0.00001" iyz="0.0000" izz="0.00001"/>
		</inertial>
	</link>
	<!--creates a (pseudo) ball joint, called from the PCC macro..-->
	<joint name="seg0_sec0-ball-ball-joint_x_joint" type="revolute">
		<parent link="softTrunk_base"/>
		<child link="seg0_sec0-ball-ball-joint_link_0"/>
		<axis xyz="1 0 0"/>
		<limit effort="1000" lower="-1.57075" upper="1.57075" velocity="1000"/>
	</joint>
	<link name="seg0_sec0-ball-ball-joint_link_0">
		<!--since the RBDL needs a inertial info for each link, just assign a really small value to links where we don't have to consider inertia.-->
		<inertial>
			<mass value="0.00001"/>
			<inertia ixx="0.00001" ixy="0.0000" ixz="0.0000" iyy="0.00001" iyz="0.0000" izz="0.00001"/>
		</inertial>
	</link>
	<joint name="seg0_sec0-ball-ball-joint_y_joint" type="revolute">
		<parent link="seg0_sec0-ball-ball-joint_link_0"/>
		<child link="seg0_sec0-ball-ball-joint_link_1"/>
		<axis xyz="0 1 0"/>
		<limit effort="1000" lower="-1.57075" upper="1.57075" velocity="1000"/>
	</joint>
	<link name="seg0_sec0-ball-ball-joint_link_1">
		<!--since the RBDL needs a inertial info for each link, just assign a really small value to links where we don't have to consider inertia.-->
		<inertial>
			<mass value="0.00001"/>
			<inertia ixx="0.00001" ixy="0.0000" ixz="0.0000" iyy="0.00001" iyz="0.0000" izz="0.00001"/>
		</inertial>
	</link>
	<joint name="seg0_sec0-ball-ball-joint_z_joint" type="revolute">
		<parent link="seg0_sec0-ball-ball-joint_link_1"/>
		<child link="seg0_sec0-a"/>
		<axis xyz="0 0 1"/>
		<limit effort="1000" lower="-1.57075" upper="1.57075" velocity="1000"/>
	</joint>
	<link name="seg0_sec0-a">
		<!--since the RBDL needs a inertial info for each link, just assign a really small value to links where we don't have to consider inertia.-->
		<inertial>
			<mass value="0.00001"/>
			<inertia ixx="0.00001" ixy="0.0000" ixz="0.0000" iyy="0.00001" iyz="0.0000" izz="0.00001"/>
		</inertial>
	</link>
	<joint name="seg0_sec0-a-b_joint" type="prismatic">
		<parent link="seg0_sec0-a"/>
		<child link="seg0_sec0-b"/>
		<axis xyz="0 0 -1"/>
		<origin xyz="0 0 0.0625"/>
		<limit effort="1000" lower="0" upper="0.0625" velocity="1000"/>
	</joint>
	<link name="seg0_sec0-b">
		<visual>
			<geometry>
				<cylinder length="0.125" radius="0.0175"/>
			</geometry>
			<!-- <origin xyz="0 0 ${length/2}"/> -->
			<material name="green"/>
		</visual>
		<visual>
			<!-- this marks the +x direction (i.e. the direction where the first chamber should be) -->
			<geometry>
				<box size="0.0035000000000000005 0.0035000000000000005 0.0035000000000000005"/>
			</geometry>
			<origin xyz="0.0175 0 0"/>
			<material name="white"/>
		</visual>
		<inertial>
			<mass value="0.16"/>
			<!-- moment of inertia for a cylinder with radius and length -->
			<inertia ixx="0.00022058333333333335" ixy="0.000" ixz="0.000" iyy="0.00022058333333333335" iyz="0.000" izz="2.4500000000000006e-05"/>
		</inertial>
	</link>
	<joint name="seg0_sec0-b-seg0_sec0-1_connect_joint" type="prismatic">
		<parent link="seg0_sec0-b"/>
		<child link="seg0_sec0-1_connect"/>
		<axis xyz="0 0 -1"/>
		<origin xyz="0 0 0.0625"/>
		<limit effort="1000" lower="0" upper="0.0625" velocity="1000"/>
	</joint>
	<link name="seg0_sec0-1_connect">
		<!--since the RBDL needs a inertial info for each link, just assign a really small value to links where we don't have to consider inertia.-->
		<inertial>
			<mass value="0.00001"/>
			<inertia ixx="0.00001" ixy="0.0000" ixz="0.0000" iyy="0.00001" iyz="0.0000" izz="0.00001"/>
		</inertial>
	</link>
	<!--creates a (pseudo) ball joint, called from the PCC macro..-->
	<joint name="seg0_sec1-ball-ball-joint_x_joint" type="revolute">
		<parent link="seg0_sec0-1_connect"/>
		<child link="seg0_sec1-ball-ball-joint_link_0"/>
		<axis xyz="1 0 0"/>
		<limit effort="1000" lower="-1.57075" upper="1.57075" velocity="1000"/>
	</joint>
	<link name="seg0_sec1-ball-ball-joint_link_0">
		<!--since the RBDL needs a inertial info for each link, just assign a really small value to links where we don't have to consider inertia.-->
		<inertial>
			<mass value="0.00001"/>
			<inertia ixx="0.00001" ixy="0.0000" ixz="0.0000" iyy="0.00001" iyz="0.0000" izz="0.00001"/>
		</inertial>
	</link>
	<joint name="seg0_sec1-ball-ball-joint_y_joint" type="revolute">
		<parent link="seg0_sec1-ball-ball-joint_link_0"/>
		<child link="seg0_sec1-ball-ball-joint_link_1"/>
		<axis xyz="0 1 0"/>
		<limit effort="1000" lower="-1.57075" upper="1.57075" velocity="1000"/>
	</joint>
	<link name="seg0_sec1-ball-ball-joint_link_1">
		<!--since the RBDL needs a inertial info for each link, just assign a really small value to links where we don't have to consider inertia.-->
		<inertial>
			<mass value="0.00001"/>
			<inertia ixx="0.00001" ixy="0.0000" ixz="0.0000" iyy="0.00001" iyz="0.0000" izz="0.00001"/>
		</inertial>
	</link>
	<joint name="seg0_sec1-ball-ball-joint_z_joint" type="revolute">
		<parent link="seg0_sec1-ball-ball-joint_link_1"/>
		<child link="seg0_sec1-a"/>
		<axis xyz="0 0 1"/>
		<limit effort="1000" lower="-1.57075" upper="1.57075" velocity="1000"/>
	</joint>
	<link name="seg0_sec1-a">
		<!--since the RBDL needs a inertial info for each link, just assign a really small value to links where we don't have to consider inertia.-->
		<inertial>
			<mass value="0.00001"/>
			<inertia ixx="0.00001" ixy="0.0000" ixz="0.0000" iyy="0.00001" iyz="0.0000" izz="0.00001"/>
		</inertial>
	</link>
	<joint name="seg0_sec1-a-b_joint" type="prismatic">
		<parent link="seg0_sec1-a"/>
		<child link="seg0_sec1-b"/>
		<axis xyz="0 0 -1"/>
		<origin xyz="0 0 0.01"/>
		<limit effort="1000" lower="0" upper="0.01" velocity="1000"/>
	</joint>
	<link name="seg0_sec1-b">
		<visual>
			<geometry>
				<cylinder length="0.02" radius="0.014"/>
			</geometry>
			<!-- <origin xyz="0 0 ${length/2}"/> -->
			<material name="green"/>
		</visual>
		<visual>
			<!-- this marks the +x direction (i.e. the direction where the first chamber should be) -->
			<geometry>
				<box size="0.0028 0.0028 0.0028"/>
			</geometry>
			<origin xyz="0.014 0 0"/>
			<material name="white"/>
		</visual>
		<inertial>
			<mass value="0.02"/>
			<!-- moment of inertia for a cylinder with radius and length -->
			<inertia ixx="1.646666666666667e-06" ixy="0.000" ixz="0.000" iyy="1.646666666666667e-06" iyz="0.000" izz="1.9600000000000003e-06"/>
		</inertial>
	</link>
	<joint name="seg0_sec1-b-seg0_sec1-2_connect_joint" type="prismatic">
		<parent link="seg0_sec1-b"/>
		<child link="seg0_sec1-2_connect"/>
		<axis xyz="0 0 -1"/>
		<origin xyz="0 0 0.01"/>
		<limit effort="1000" lower="0" upper="0.01" velocity="1000"/>
	</joint>
	<link name="seg0_sec1-2_connect">
		<!--since the RBDL needs a inertial info for each link, just assign a really small value to links where we don't have to consider inertia.-->
		<inertial>
			<mass value="0.00001"/>
			<inertia ixx="0.00001" ixy="0.0000" ixz="0.0000" iyy="0.00001" iyz="0.0000" izz="0.00001"/>
		</inertial>
	</link>
	<!--creates a (pseudo) ball joint, called from the PCC macro..-->
	<joint name="seg1_sec0-ball-ball-joint_x_joint" type="revolute">
		<parent link="seg0_sec1-2_connect"/>
		<child link="seg1_sec0-ball-ball-joint_link_0"/>
		<axis xyz="1 0 0"/>
		<limit effort="1000" lower="-1.57075" upper="1.57075" velocity="1000"/>
	</joint>
	<link name="seg1_sec0-ball-ball-joint_link_0">
		<!--since the RBDL needs a inertial info for each link, just assign a really small value to links where we don't have to consider inertia.-->
		<inertial>
			<mass value="0.00001"/>
			<inertia ixx="0.00001" ixy="0.0000" ixz="0.0000" iyy="0.00001" iyz="0.0000" izz="0.00001"/>
		</inertial>
	</link>
	<joint name="seg1_sec0-ball-ball-joint_y_joint" type="revolute">
		<parent link="seg1_sec0-ball-ball-joint_link_0"/>
		<child link="seg1_sec0-ball-ball-joint_link_1"/>
		<axis xyz="0 1 0"/>
		<limit effort="1000" lower="-1.57075" upper="1.57075" velocity="1000"/>
	</joint>
	<link name="seg1_sec0-ball-ball-joint_link_1">
		<!--since the RBDL needs a inertial info for each link, just assign a really small value to links where we don't have to consider inertia.-->
		<inertial>
			<mass value="0.00001"/>
			<inertia ixx="0.00001" ixy="0.0000" ixz="0.0000" iyy="0.00001" iyz="0.0000" izz="0.00001"/>
		</inertial>
	</link>
	<joint name="seg1_sec0-ball-ball-joint_z_joint" type="revolute">
		<parent link="seg1_sec0-ball-ball-joint_link_1"/>
		<child link="seg1_sec0-a"/>
		<axis xyz="0 0 1"/>
		<limit effort="1000" lower="-1.57075" upper="1.57075" velocity="1000"/>
	</joint>
	<link name="seg1_sec0-a">
		<!--since the RBDL needs a inertial info for each link, just assign a really small value to links where we don't have to consider inertia.-->
		<inertial>
			<mass value="0.00001"/>
			<inertia ixx="0.00001" ixy="0.0000" ixz="0.0000" iyy="0.00001" iyz="0.0000" izz="0.00001"/>
		</inertial>
	</link>
	<joint name="seg1_sec0-a-b_joint" type="prismatic">
		<parent link="seg1_sec0-a"/>
		<child link="seg1_sec0-b"/>
		<axis xyz="0 0 -1"/>
		<origin xyz="0 0 0.0625"/>
		<limit effort="1000" lower="0" upper="0.0625" velocity="1000"/>
	</joint>
	<link name="seg1_sec0-b">
		<visual>
			<geometry>
				<cylinder length="0.125" radius="0.014"/>
			</geometry>
			<!-- <origin xyz="0 0 ${length/2}"/> -->
			<material name="green"/>
		</visual>
		<visual>
			<!-- this marks the +x direction (i.e. the direction where the first chamber should be) -->
			<geometry>
				<box size="0.0028 0.0028 0.0028"/>
			</geometry>
			<origin xyz="0.014 0 0"/>
			<material name="white"/>
		</visual>
		<inertial>
			<mass value="0.082"/>
			<!-- moment of inertia for a cylinder with radius and length -->
			<inertia ixx="0.00011078883333333335" ixy="0.000" ixz="0.000" iyy="0.00011078883333333335" iyz="0.000" izz="8.036e-06"/>
		</inertial>
	</link>
	<joint name="seg1_sec0-b-seg1_sec0-1_connect_joint" type="prismatic">
		<parent link="seg1_sec0-b"/>
		<child link="seg1_sec0-1_connect"/>
		<axis xyz="0 0 -1"/>
		<origin xyz="0 0 0.0625"/>
		<limit effort="1000" lower="0" upper="0.0625" velocity="1000"/>
	</joint>
	<link name="seg1_sec0-1_connect">
		<!--since the RBDL needs a inertial info for each link, just assign a really small value to links where we don't have to consider inertia.-->
		<inertial>
			<mass value="0.00001"/>
			<inertia ixx="0.00001" ixy="0.0000" ixz="0.0000" iyy="0.00001" iyz="0.0000" izz="0.00001"/>
		</inertial>
	</link>
	<!--creates a (pseudo) ball joint, called from the PCC macro..-->
	<joint name="seg1_sec1-ball-ball-joint_x_joint" type="revolute">
		<parent link="seg1_sec0-1_connect"/>
		<child link="seg1_sec1-ball-ball-joint_link_0"/>
		<axis xyz="1 0 0"/>
		<limit effort="1000" lower="-1.57075" upper="1.57075" velocity="1000"/>
	</joint>
	<link name="seg1_sec1-ball-ball-joint_link_0">
		<!--since the RBDL needs a inertial info for each link, just assign a really small value to links where we don't have to consider inertia.-->
		<inertial>
			<mass value="0.00001"/>
			<inertia ixx="0.00001" ixy="0.0000" ixz="0.0000" iyy="0.00001" iyz="0.0000" izz="0.00001"/>
		</inertial>
	</link>
	<joint name="seg1_sec1-ball-ball-joint_y_joint" type="revolute">
		<parent link="seg1_sec1-ball-ball-joint_link_0"/>
		<child link="seg1_sec1-ball-ball-joint_link_1"/>
		<axis xyz="0 1 0"/>
		<limit effort="1000" lower="-1.57075" upper="1.57075" velocity="1000"/>
	</joint>
	<link name="seg1_sec1-ball-ball-joint_link_1">
		<!--since the RBDL needs a inertial info for each link, just assign a really small value to links where we don't have to consider inertia.-->
		<inertial>
			<mass value="0.00001"/>
			<inertia ixx="0.00001" ixy="0.0000" ixz="0.0000" iyy="0.00001" iyz="0.0000" izz="0.00001"/>
		</inertial>
	</link>
	<joint name="seg1_sec1-ball-ball-joint_z_joint" type="revolute">
		<parent link="seg1_sec1-ball-ball-joint_link_1"/>
		<child link="seg1_sec1-a"/>
		<axis xyz="0 0 1"/>
		<limit effort="1000" lower="-1.57075" upper="1.57075" velocity="1000"/>
	</joint>
	<link name="seg1_sec1-a">
		<!--since the RBDL needs a inertial info for each link, just assign a really small value to links where we don't have to consider inertia.-->
		<inertial>
			<mass value="0.00001"/>
			<inertia ixx="0.00001" ixy="0.0000" ixz="0.0000" iyy="0.00001" iyz="0.0000" izz="0.00001"/>
		</inertial>
	</link>
	<joint name="seg1_sec1-a-b_joint" type="prismatic">
		<parent link="seg1_sec1-a"/>
		<child link="seg1_sec1-b"/>
		<axis xyz="0 0 -1"/>
		<origin xyz="0 0 0.01"/>
		<limit effort="1000" lower="0" upper="0.01" velocity="1000"/>
	</joint>
	<link name="seg1_sec1-b">
		<visual>
			<geometry>
				<cylinder length="0.02" radius="0.0099"/>
			</geometry>
			<!-- <origin xyz="0 0 ${length/2}"/> -->
			<material name="green"/>
		</visual>
		<visual>
			<!-- this marks the +x direction (i.e. the direction where the first chamber should be) -->
			<geometry>
				<box size="0.00198 0.00198 0.00198"/>
			</geometry>
			<origin xyz="0.0099 0 0"/>
			<material name="white"/>
		</visual>
		<inertial>
			<mass value="0.023"/>
			<!-- moment of inertia for a cylinder with radius and length -->
			<inertia ixx="1.3302241666666666e-06" ixy="0.000" ixz="0.000" iyy="1.3302241666666666e-06" iyz="0.000" izz="1.1271150000000002e-06"/>
		</inertial>
	</link>
	<joint name="seg1_sec1-b-seg1_sec1-2_connect_joint" type="prismatic">
		<parent link="seg1_sec1-b"/>
		<child link="seg1_sec1-2_connect"/>
		<axis xyz="0 0 -1"/>
		<origin xyz="0 0 0.01"/>
		<limit effort="1000" lower="0" upper="0.01" velocity="1000"/>
	</joint>
	<link name="seg1_sec1-2_connect">
		<!--since the RBDL needs a inertial info for each link, just assign a really small value to links where we don't have to consider inertia.-->
		<inertial>
			<mass value="0.00001"/>
			<inertia ixx="0.00001" ixy="0.0000" ixz="0.0000" iyy="0.00001" iyz="0.0000" izz="0.00001"/>
		</inertial>
	</link>
</robot>
